### PR TITLE
Add registration methods for a poller

### LINKED ISSUE #11: Setup automatic registration of the Client Poller HealthChecks

### PR DESCRIPTION
**Pull request by chrisrohr**
Closes #11

### CODE REVIEW

File: src/main/java/org/kiwiproject/dropwizard/poller/health/ClientPollerHealthChecks.java [hunk 1]
```diff
@@ -96,4 +104,75 @@ public static Result unhealthy(ClientPollerStatistics statistics, HealthStatus s
...
+     *
+     * @param poller The poller being monitored
+     * @return a list of health checks for the poller
+     */
+    public static List<PollerHealthCheck> buildPollerHealthChecks(ClientPoller poller) {
+        return buildPollerHealthChecks(poller, PollerHealthCheckConfig.builder().build());
+    }
+
+    /**
+     * Build the standard client poller health checks, returning a list containing the actual health checks wrapped in a
```
Comment (sleberknight) [inline src/main/java/org/kiwiproject/dropwizard/poller/health/ClientPollerHealthChecks.java:153]: Should we state explicitly that the returned list is _mutable_? Or, should we actually be returning an unmodifiable list, e.g. `List.of`?

File: src/main/java/org/kiwiproject/dropwizard/poller/health/ClientPollerHealthChecks.java [hunk 2]
```diff
@@ -96,4 +104,75 @@ public static Result unhealthy(ClientPollerStatistics statistics, HealthStatus s
...
+
+    /**
+     * Convenience method that registers the following health checks for the given poller and Dropwizard environment:
+     * {@link ClientPollerTimeBasedHealthCheck}, {@link ClientPollerLatencyBasedHealthCheck}, and
+     * {@link ClientPollerMissedPollHealthCheck}
+     *
+     * @param poller      The poller being monitored
+     * @param environment The dropwizard environment
+     */
+    public static void registerPollerHealthChecks(ClientPoller poller, Environment environment) {
```
Comment (sleberknight) [inline src/main/java/org/kiwiproject/dropwizard/poller/health/ClientPollerHealthChecks.java:116]: Is there any reason we might want to return a list containing the registered health checks?

File: src/main/java/org/kiwiproject/dropwizard/poller/health/ClientPollerHealthChecks.java [hunk 3]
```diff
@@ -96,4 +104,75 @@ public static Result unhealthy(ClientPollerStatistics statistics, HealthStatus s
...
+    /**
+     * Convenience method that registers the following health checks for the given poller and Dropwizard environment:
+     * {@link ClientPollerTimeBasedHealthCheck}, {@link ClientPollerLatencyBasedHealthCheck}, and
+     * {@link ClientPollerMissedPollHealthCheck}
+     *
+     * @param poller            The poller being monitored
+     * @param environment       The dropwizard environment
+     * @param healthCheckConfig The config for the health checks
+     */
+    public static void registerPollerHealthChecks(ClientPoller poller,
```
Comment (sleberknight) [inline src/main/java/org/kiwiproject/dropwizard/poller/health/ClientPollerHealthChecks.java:129]: Ditto above question on returning list of registered health checks.

**Review by sleberknight — APPROVED** [review]
All comments are minor and can be safely ignored unless you think they are worth considering.

File: src/main/java/org/kiwiproject/dropwizard/poller/health/ClientPollerHealthChecks.java (hunk 1 above)
Comment (chrisrohr) [reply src/main/java/org/kiwiproject/dropwizard/poller/health/ClientPollerHealthChecks.java]: I made it List.of

File: src/main/java/org/kiwiproject/dropwizard/poller/health/ClientPollerHealthChecks.java (hunk 2 above)
Comment (chrisrohr) [reply src/main/java/org/kiwiproject/dropwizard/poller/health/ClientPollerHealthChecks.java]: Reworked this

**Review by sleberknight — APPROVED** [review]

**Pull request merged**